src/irr/asset/CGLSLBRDFBuiltinIncludeLoader.h

```diff
@@ -284,6 +278,53 @@ vec3 irr_glsl_blinn_phong_fresnel_conductor_cos_eval(in irr_glsl_BSDFIsotropicPa
     return irr_glsl_blinn_phong(params.NdotH, n) * irr_glsl_fresnel_conductor(ior2[0], ior2[1], params.VdotH) / denom;
 }
 
+#endif
+)";
+    }
+    static std::string getAshikhminShirleyNDF(const std::string&)
+    {
+        return
+R"(#ifndef _IRR_BSDF_BRDF_SPECULAR_NDF_ASHIKHMIN_SHIRLEY_INCLUDED_
+#define _IRR_BSDF_BRDF_SPECULAR_NDF_ASHIKHMIN_SHIRLEY_INCLUDED_
+
+#include <irr/builtin/glsl/bsdf/common.glsl>
+
+//n is 2 phong-like exponents for anisotropy, can be defined as vec2(1.0/at, 1.0/ab) where at is roughness along tangent direction and ab is roughness along bitangent direction
+//sin_cos_phi is sin and cos of azimuth angle of half vector
+float irr_glsl_ashikhmin_shirley(in float NdotL, in float NdotV, in float NdotH, in float VdotH, in vec2 n, in vec2 sin_cos_phi)
+{
+    float nom = sqrt((n.x + 1.0)*(n.y + 1.0)) * pow(NdotH, n.x*sin_cos_phi.x*sin_cos_phi.x + n.y*sin_cos_phi.y*sin_cos_phi.y);
+    float denom = 8.0 * irr_glsl_PI * VdotH * max(NdotV,NdotL);
+
+    return NdotL * nom/denom;
+}
+
+#endif
+)";
+    }
+    static std::string getAshikhminShirley_cos_eval(const std::string&)
+    {
+        return
+R"(#ifndef _IRR_BSDF_BRDF_SPECULAR_NDF_ASHIKHMIN_SHIRLEY_INCLUDED_
+#define _IRR_BSDF_BRDF_SPECULAR_NDF_ASHIKHMIN_SHIRLEY_INCLUDED_
+
+#include <irr/builtin/glsl/bsdf/common.glsl>
+#include <irr/builtin/glsl/bsdf/brdf/specular/ndf/ashikhmin_shirley.glsl>
+#include <irr/builtin/glsl/bsdf/brdf/specular/fresnel/fresnel.glsl>
+
+//n is 2 phong-like exponents for anisotropy, can be defined as vec2(1.0/at, 1.0/ab) where at is roughness along tangent direction and ab is roughness along bitangent direction
+//sin_cos_phi is sin and cos of azimuth angle of half vector
+float irr_glsl_ashikhmin_shirley_cos_eval(in irr_glsl_BSDFAnisotropicParams params, in vec2 n, in vec2 sin_cos_phi, in mat2x3 ior2)
+{
+    float ndf = irr_glsl_ashikhmin_shirley(params.isotropic.NdotL, params.isotropic.NdotV, params.isotropic.NdotH, params.isotropic.VdotH, n, sin_cos_phi);
+    float fr = irr_glsl_fresnel_conductor(ior2[0], ior2[1], params.isotropic.VdotH);
+
+    //shadowing/masking term missing, actually ashikhmin-shirley model was designed with an idea that G term is almost irrelevant, 
```

Comment:
ok, their shadowing term is cumbersome because it requires precomputing integrals (if curious see section 4.2, especially `g(k)`, and section 5.2). However it's very general and apparently can be used with any NDF, so might get useful some day

Reply:
we don't care about ahikhmin-shirley enough to devote time to this, just slap on the anisotropic GGX shadowing and make a note that its wrong